archive/android_bootldr.ksy

```diff
@@ -0,0 +1,36 @@
+meta:
+  id: android_bootldr
+  title: Qualcomm Snapdragon (MSM) bootloader.img format
```

Comment:
BTW, the Wikipedia article [Devices using Qualcomm Snapdragon processors](https://en.wikipedia.org/wiki/Devices_using_Qualcomm_Snapdragon_processors) lists devices grouped by Qualcomm model number, so it may be helpful for finding out which devices will ultimately support and use this format (by checking if the model number starts with `MSM`, e.g. `MSM8996`). You might want to include this link into `/doc` (please don't forget about these angle brackets `<>`, see https://github.com/kaitai-io/kaitai_struct_formats/pull/467#discussion_r609723301).

Reply:
I will search for information. Not every device with a Snapdragon used it, just a few did. Welcome to the mess that is Android! ;-)

Reply:
> Not every device with a Snapdragon used it, just a few did.

Yes, but I downloaded and inspected a few factory images from https://developers.google.com/android/images and all models claimed to have the `MSM*` Snapdragon processor that I checked actually did use this format, so it seems to be correlated:

* ["hammerhead" for Nexus 5](https://developers.google.com/android/images#hammerhead): ✔️ (processor `MSM8974AA`, see [wiki](https://en.wikipedia.org/wiki/Devices_using_Qualcomm_Snapdragon_processors#Snapdragon_800,_801_and_805_(2013/14)))
* ["bullhead" for Nexus 5X](https://developers.google.com/android/images#bullhead): ✔️ (processor `MSM8992`, see [wiki](https://en.wikipedia.org/wiki/Devices_using_Qualcomm_Snapdragon_processors#Snapdragon_808_and_810_(2015)))
* ["sailfish" for Pixel](https://developers.google.com/android/images#sailfish): ✔️ (processor `MSM8996 Pro-AB`, see [wiki](https://en.wikipedia.org/wiki/Devices_using_Qualcomm_Snapdragon_processors#Snapdragon_820_and_821_(2016)))
* ["sargo" for Pixel 3a](https://developers.google.com/android/images#sargo): ❌ (processor `SDM670`, see [wiki](https://en.wikipedia.org/wiki/Devices_using_Qualcomm_Snapdragon_processors#Snapdragon_632_and_670_(2018)))
* ["redfin" for Pixel 5](https://developers.google.com/android/images#redfin): ❌ (processor `SM7250-AB`, see [wiki](https://en.wikipedia.org/wiki/Devices_using_Qualcomm_Snapdragon_processors#Snapdragon_750G_and_765/765G/768G_5G_(2020)))

> Welcome to the mess that is Android! ;-)

You're certainly right about that :)

Reply:
I have added a short description, although not an in depth one.